### PR TITLE
Replace integer check with Number.isInteger

### PR DESCRIPTION
**Pull request by mAAdhaTTah**
This works correctly for large numbers.

Related to https://github.com/jquense/yup/pull/147

### CODE REVIEW

**Comment by mAAdhaTTah** [comment]
I'm not sure what your policy is for using newer features. I can inline the `Number.isInteger` polyfill from MDN if that's preferred. Or I can update the docs to indicate this is required so the user makes sure it's supported or is polyfilled.

**Comment by mAAdhaTTah** [comment]
Looking at the test failure, I'm not sure why that was considered invalid.

File: src/number.js [hunk 1]
```diff
@@ -94,7 +92,7 @@ inherits(NumberSchema, MixedSchema, {
   },
 
   integer(message = locale.integer) {
-    return this.test({ name: 'integer', message, test: isInteger });
+    return this.test({ name: 'integer', message, test: Number.isInteger });
```
Comment (jquense) [inline src/number.js]: this still needs to account for missing values and return `true` in those cases

**Comment by mAAdhaTTah** [comment]
Addressed the comment & resolved the conflict.

**Comment by mAAdhaTTah** [comment]
@jquense Bumping this. I think I've addressed your issues. Let me know if you've got other changes.

**Comment by jquense** [comment]
thanks!

**Comment by pleunv** [comment]
Just discovered that `Number.isInteger(Number.MAX_SAFE_INTEGER + 1)` returns `true`, which means that this could cause issues in validation. There's also `Number.isSafeInteger()` which would invalidate anything higher than `Number.MAX_SAFE_INTEGER`. Should Yup be using this one instead?

**Pull request merged**